evals/api.py

```diff
@@ -23,85 +24,139 @@
 logger = logging.getLogger(__name__)
 
 
-def completion_query(
-    model_spec: ModelSpec,
-    prompt: Union[OpenAICreatePrompt, OpenAICreateChatPrompt, Prompt],
-    **kwargs,
-) -> tuple[dict, Union[OpenAICreatePrompt, OpenAICreateChatPrompt], dict]:
-    """
-    Query the API for a completion.
+class ReturnType(ABC):
+    def __init__(self, raw_data: Any, prompt: Any, metadata: Dict[str, Any] = None):
+        self.raw_data = raw_data
+        self.prompt: str = prompt
+        self.completions: List[str] = self.extract_completions()
+        self.metadata = metadata if metadata is not None else {}
 
-    ARGS
-    ====
-    `model_spec`: `ModelSpec` containing model details to use in the query.
-        This should be the dict returned by `registry.get_model()`.
-        If `model_spec` is not provided, we use the default model that was
-            intialized at the beginning of the run.
-    `prompt`: Either a `Prompt` object or a raw prompt that will get wrapped in
-        the approriate `Prompt` class.
-    `kwargs`: Other arguments passed to the API.
+    @abstractmethod
+    def extract_completions(self) -> List[str]:
```

Comment:
What about this instead?

```
@property # or maybe @cached_property?
def completions(self) -> List[str]
```

so that you can use it like

```
result: ReturnType = ..
completion = result.completions[0]
```

Alternatively, we could keep as is (maybe rename to `get_completions` since it's shorter)